sympy/matrices/expressions/hadamard.py

```diff
@@ -311,6 +320,16 @@ def _eval_transpose(self):
         from sympy.matrices.expressions.transpose import transpose
         return HadamardPower(transpose(self.base), self.exp)
 
+    def _eval_derivative(self, x):
+        from sympy import log
+        dexp = self.exp.diff(x)
+        logbase = self.base.applyfunc(log)
+        dlbase = logbase.diff(x)
+        return hadamard_product(
+            dexp*logbase + self.exp*dlbase,
+            self
+        )
```

Comment:
This should be the same as `Pow._eval_derivative`, maybe it can be shared in the future.